Sources/Core/Microsoft.StreamProcessing/Pipes/BinaryPlanNode.cs

```diff
@@ -159,10 +159,15 @@ public enum JoinKind
         Clip,
 
         /// <summary>
-        /// An equijoin between two stream.
+        /// An equijoin between two streams.
         /// </summary>
         EquiJoin,
 
+        /// <summary>
+        /// An equijoin where both streams are guaranteed to only contain intervals.
```

Comment:
>only contain intervals. [](start = 61, length = 23)

only contain constant-duration intervals

Reply:
Regarding the Traverser, I looked at consolidating them, and they seemed different enough that unifying them wasn't straightforward. I'll give it another look though just in case.